Source/Renderer/Context.js

```diff
@@ -269,24 +284,55 @@ define([
         this._antialias = gl.getContextAttributes().antialias;
 
         // Query and initialize extensions
-        this._standardDerivatives = getExtension(gl, ['OES_standard_derivatives']);
-        this._elementIndexUint = getExtension(gl, ['OES_element_index_uint']);
-        this._depthTexture = getExtension(gl, ['WEBGL_depth_texture', 'WEBKIT_WEBGL_depth_texture']);
-        this._textureFloat = getExtension(gl, ['OES_texture_float']);
+        this._standardDerivatives = !!getExtension(gl, ['OES_standard_derivatives']);
+        this._elementIndexUint = !!getExtension(gl, ['OES_element_index_uint']);
+        this._depthTexture = !!getExtension(gl, ['WEBGL_depth_texture', 'WEBKIT_WEBGL_depth_texture']);
+        this._textureFloat = !!getExtension(gl, ['OES_texture_float']);
+        this._fragDepth = !!getExtension(gl, ['EXT_frag_depth']);
+        this._debugShaders = getExtension(gl, ['WEBGL_debug_shaders']);
 
         var textureFilterAnisotropic = options.allowTextureFilterAnisotropic ? getExtension(gl, ['EXT_texture_filter_anisotropic', 'WEBKIT_EXT_texture_filter_anisotropic']) : undefined;
-        this._textureFilterAnisotropic = textureFilterAnisotropic;
+        this._textureFilterAnisotropic = !!textureFilterAnisotropic;
         ContextLimits._maximumTextureFilterAnisotropy = defined(textureFilterAnisotropic) ? gl.getParameter(textureFilterAnisotropic.MAX_TEXTURE_MAX_ANISOTROPY_EXT) : 1.0;
+        
+        if (webgl2) {
```

Comment:
I'd rather only assign to each context property once since I'm not 100% sure that all JavaScript engines will optimize (not enter dictionary mode) when we "create" properties in a branch (even though it is in both branches).

Instead, perhaps something like:

```javascript
var glCreateVertexArray = undefined;

if (webgl2) {
   glCreateVertexArray = // ...
} else {
   glCreateVertexArray = // ...
}

this.glCreateVertexArray = glCreateVertexArray;
```

Or leave it if someone can confirm the performance.